tests/layer_tests/tensorflow_tests/test_tf_Equal.py

```diff
@@ -0,0 +1,85 @@
+# Copyright (C) 2018-2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import numpy as np
+import pytest
+import tensorflow as tf
+from common.tf_layer_test_class import CommonTFLayerTest
+from common.utils.tf_utils import permute_nchw_to_nhwc
+
+import logging
+
+# Testing operation Equal
+# Documentation: https://www.tensorflow.org/versions/r1.15/api_docs/python/tf/math/equal
+
+class TestTFEqual(CommonTFLayerTest):
+    equal_output_type_param = np.float32
+    is_equation_test = False
+
+    # Overload inputs generation to fill dummy Equal input with 0
+    def _prepare_input(self, inputs_dict):
+        if self.is_equation_test == True:
+            for input in inputs_dict.keys():
+                inputs_dict[input] = np.zeros(inputs_dict[input]).astype(self.equal_output_type_param)
+            return inputs_dict
+        else:
+            return CommonTFLayerTest._prepare_input(self, inputs_dict)
+
+
+    def create_tf_equal_net(self, x_shape, y_shape, equation_test, ir_version, use_new_frontend):
+        self.is_equation_test = equation_test
+
+        tf.compat.v1.reset_default_graph()
+
+        # Create the graph and model
+        with tf.compat.v1.Session() as sess:
+            tf_x_shape = x_shape.copy()
+            tf_y_shape = y_shape.copy()
+
+            # I'm not sure it is necessary, have to check
+            tf_x_shape = permute_nchw_to_nhwc(tf_x_shape, use_new_frontend)
+            tf_y_shape = permute_nchw_to_nhwc(tf_y_shape, use_new_frontend)
+
+            x = tf.compat.v1.placeholder(tf.float32, tf_x_shape, 'Input')
+            if self.is_equation_test == True:
+                constant_value = np.zeros(tf_y_shape).astype(self.equal_output_type_param)
+            else:
+                # Why range -256..256 was chosen - no ideas
+                constant_value = np.random.randint(-256, 256, tf_y_shape).astype(self.equal_output_type_param)
+                if (constant_value == 0).all():
+                    # Avoid elimination of the layer from IR
+                    constant_value = constant_value + 1
+
+            y = tf.constant(constant_value)
+
+            #It doesn't dump exact matrix as I expect
+            logging.getLogger().info(f"{x}")
+            logging.getLogger().info(f"{y}")
+
+            equal = tf.equal(x, y)
+
+            tf.compat.v1.global_variables_initializer()
+            tf_net = sess.graph_def
+
+        ref_net = None
+        return tf_net, ref_net
+
+    test_data = [
+        dict(x_shape=[1], y_shape=[1], equation_test=False),
+        dict(x_shape=[1], y_shape=[1], equation_test=True),
+        dict(x_shape=[1,1,1], y_shape=[1], equation_test=False),
+        dict(x_shape=[1], y_shape=[1,1,1], equation_test=False),
+        dict(x_shape=[1,1,1,1], y_shape=[1,1,1,1], equation_test=False),
+        dict(x_shape=[1,1,1,1], y_shape=[1,1,1,1], equation_test=True),
+        ]
+
```

Comment:
please add separately test cases for fp32 and int32 inputs.
For int32, you can randomly generate elements from the short range, for example from [-3, 3], this way you can provide both `True` and `False` elements in the resulted tensor.

please use not input tensors not only with the single element but also input tensors should be of [2, 3] shape, [4,3,2], for example.

Reply:
Moreover, int32 case for Equal is the most frequent case in the models because it used for integer masks and indices

Reply:
Also, please add one case to the pre-commit